### ISSUE TITLE
Encountered a Dart Analysis Server error

### ISSUE THREAD

**Issue by jaybryantc** [dart-lang/sdk#38079]
Analyzer Feedback from IntelliJ

## Version information

- `IDEA AI-191.8026.42.35.5791312`
- `2.4.0`
- `AI-191.8026.42.35.5791312, JRE 1.8.0_202-release-1483-b49-5587405x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.14.6, screens 1440x900, 1920x1080  Retina`

## Exception
```
Dart analysis server, SDK version 2.4.0, server version 1.27.1, error: Analysis failed: /Users/jaybryantc/Documents/ffuf/kemmlit-kemmon-flutter/lib/pages/complaint_page.dart context: exception_20190828_153017_443
NoSuchMethodError: The getter 'element' was called on null.
Receiver: null
Tried calling: element
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:50:5)
#1      StaticTypeAnalyzer._findIteratedType._find (package:analyzer/src/generated/static_type_analyzer.dart:1333:35)
#2      StaticTypeAnalyzer._findIteratedType._find (package:analyzer/src/generated/static_type_analyzer.dart:1351:21)
#3      _ListBase&Object&ListMixin.any (dart:collection/list.dart:125:15)
#4      StaticTypeAnalyzer._findIteratedType._find (package:analyzer/src/generated/static_type_analyzer.dart:1353:25)
#5      StaticTypeAnalyzer._findIteratedType._find (package:analyzer/src/generated/static_type_analyzer.dart:1351:21)
#6      StaticTypeAnalyzer._findIteratedType._find (package:analyzer/src/generated/static_type_analyzer.dart:1351:21)
#7      StaticTypeAnalyzer._findIteratedType._find (package:analyzer/src/generated/static_type_analyzer.dart:1351:21)
#8      StaticTypeAnalyzer._findIteratedType._find (package:analyzer/src/generated/static_type_analyzer.dart:1351:21)
#9      StaticTypeAnalyzer._findIteratedType (package:analyzer/src/generated/static_type_analyzer.dart:1360:12)
#10     StaticTypeAnalyzer._inferForEachLoopVariableType (package:analyzer/src/generated/static_type_analyzer.dart:1580:33)
#11     StaticTypeAnalyzer.visitDeclaredIdentifier (package:analyzer/src/generated/static_type_analyzer.dart:459:5)
#12     DeclaredIdentifierImpl.accept (package:analyzer/src/dart/ast/ast.dart:2954:49)
#13     ResolverVisitor.visitNode (package:analyzer/src/generated/resolver.dart:4772:10)
#14     UnifyingAstVisitor.visitDeclaredIdentifier (package:analyzer/dart/ast/visitor.dart:3168:57)
#15     ScopedVisitor.visitDeclaredIdentifier (package:analyzer/src/generated/resolver.dart:5804:11)
#16     DeclaredIdentifierImpl.accept (package:analyzer/src/dart/ast/ast.dart:2954:49)
#17     ResolverVisitor.visitForStatementInScope (package:analyzer/src/generated/resolver.dart:4495:21)
#18     ScopedVisitor.visitForStatement (package:analyzer/src/generated/resolver.dart:5904:7)
#19     _UnifyingAstVisitor&Object&UIAsCodeVisitorMixin.visitForStatement2 (package:analyzer/src/dart/ast/utilities.dart:9250:12)
#20     ForStatement2Impl.accept (package:analyzer/src/dart/ast/ast.dart:4653:49)
#21     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7676:20)
#22     BlockImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:1121:17)
#23     ResolverVisitor.visitNode (package:analyzer/src/generated/resolver.dart:4770:10)
#24     UnifyingAstVisitor.visitBlock (package:analyzer/dart/ast/visitor.dart:3116:31)
#25     ScopedVisitor.visitBlock (package:analyzer/src/generated/resolver.dart:5658:13)
#26     BlockImpl.accept (package:analyzer/src/dart/ast/ast.dart:1117:49)
#27     BlockFunctionBodyImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:1076:13)
#28     ResolverVisitor.visitNode (package:analyzer/src/generated/resolver.dart:4770:10)
#29     UnifyingAstVisitor.visitBlockFunctionBody (package:analyzer/dart/ast/visitor.dart:3119:55)
#30     ScopedVisitor.visitBlockFunctionBody (package:analyzer/src/generated/resolver.dart:5669:13)
#31     ResolverVisitor.visitBlockFunctionBody (package:analyzer/src/generated/resolver.dart:4114:13)
#32     BlockFunctionBodyImpl.accept (package:analyzer/src/dart/ast/ast.dart:1072:49)
#33     MethodDeclarationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:7116:12)
#34     ResolverVisitor.visitNode (package:analyzer/src/generated/resolver.dart:4770:10)
#35     UnifyingAstVisitor.visitMethodDeclaration (package:analyzer/dart/ast/visitor.dart:3321:55)
#36     ScopedVisitor.visitMethodDeclarationInScope (package:analyzer/src/generated/resolver.dart:6112:11)
#37     ResolverVisitor.visitMethodDeclarationInScope (package:analyzer/src/generated/resolver.dart:4720:11)
#38     ScopedVisitor.visitMethodDeclaration (package:analyzer/src/generated/resolver.dart:6105:7)
#39     ResolverVisitor.visitMethodDeclaration (package:analyzer/src/generated/resolver.dart:4711:13)
#40     MethodDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:7107:49)
#41     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7676:20)
#42     ScopedVisitor.visitClassMembersInScope (package:analyzer/src/generated/resolver.dart:5735:18)
#43     ScopedVisitor.visitClassDeclaration (package:analyzer/src/generated/resolver.dart:5713:11)
#44     ResolverVisitor.visitClassDeclaration (package:analyzer/src/generated/resolver.dart:4150:13)
#45     ClassDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:1560:49)
#46     ResolverVisitor.visitCompilationUnit (package:analyzer/src/generated/resolver.dart:4209:23)
#47     CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2146:49)
#48     LibraryAnalyzer._resolveFile (package:analyzer/src/dart/analysis/library_analyzer.dart:692:10)
#49     Li...
```

For additional log information, please append the contents of
file:///private/var/folders/pj/c78ch1vx7kvdn9mcx_s9k1jc0000gp/T/report.txt.

**Comment by srawlins** [comment]
Looks super similar to https://github.com/dart-lang/sdk/issues/36124, but not exactly the  same stacktrace...

**Comment by srawlins** [comment]
I suspect this has been fixed, I cannot recreate a related crash in 2.5.0-dev.2.1. Please re-open if you encounter this again.
